scripts/generate_driver_wrappers.py

```diff
@@ -22,52 +22,153 @@
 
     with open(os.path.join(output_dir, "psa_crypto_driver_wrappers.c"), 'w') as out_file:
         out_file.write(result)
 
+
+def validate_json(driverjson_data: Driver, driverschema_list: dict) -> bool:
+    """
+    Validate the Driver JSON against an appropriate schema
+    the schema passed could be that matching an opaque/ transparent driver.
+    """
+    driver_type = driverjson_data["type"]
+    driver_prefix = driverjson_data["prefix"]
+    try:
+        _schema = driverschema_list[driver_type]
+        jsonschema.validate(instance=driverjson_data, schema=_schema)
+
+    except KeyError as err:
+        # This could happen if the driverjson_data.type does not exist in the passed in schema list
+        # schemas = {'transparent': transparent_driver_schema, 'opaque': opaque_driver_schema}
+        # Print onto stdout and stderr.
+        print("Unknown Driver type " + driver_type +
+              " for driver " + driver_prefix, str(err))
+        print("Unknown Driver type " + driver_type +
+              " for driver " + driver_prefix, str(err), file=sys.stderr)
+        return False
+
+    except jsonschema.exceptions.ValidationError as err:
+        # Print onto stdout and stderr.
+        print("Error: Failed to validate data file: {} using schema: {}."
+              "\n Exception Message: \"{}\""
+              " ".format(driverjson_data, _schema, str(err)))
+        print("Error: Failed to validate data file: {} using schema: {}."
+              "\n Exception Message: \"{}\""
+              " ".format(driverjson_data, _schema, str(err)), file=sys.stderr)
+        return False
+
+    return True
+
+def read_driver_descriptions(mbedtls_root: str, json_directory: str, \
+                          jsondriver_list: str) -> Tuple[bool, list]:
+    """
+    Merge driver JSON files into a single ordered JSON after validation.
+    """
+    result = []
+    with open(os.path.join(mbedtls_root,
+                           'scripts',
+                           'data_files',
+                           'driver_jsons',
+                           'driver_transparent_schema.json'), 'r') as file:
+        transparent_driver_schema = json.load(file)
+    with open(os.path.join(mbedtls_root,
+                           'scripts',
+                           'data_files',
+                           'driver_jsons',
+                           'driver_opaque_schema.json'), 'r') as file:
+        opaque_driver_schema = json.load(file)
+
+    driver_schema_list = {'transparent':transparent_driver_schema,
+                          'opaque':opaque_driver_schema}
+
+    with open(os.path.join(json_directory, jsondriver_list), 'r') as driverlistfile:
+        driverlist = json.load(driverlistfile)
+    for file_name in driverlist:
+        with open(os.path.join(json_directory, file_name), 'r') as infile:
+            json_data = json.load(infile)
+            ret = validate_json(json_data, driver_schema_list)
+            if ret is False:
```

Comment:
Since that is exitting the following loop: `for file_name in driverlist:`, I assume that `read_driver_descriptions()` is by design intended to return `(False, [])` if a single of the driver input files does not validate? 

If that is the intended design choice, no action is required. If we want to just skip the file that fails to validate, we could replace that with `continue`

Reply:
Yes, If any driver fails we intend on failing.
We don't have the tooling to fail certain drivers  while syntactically correct schemas continue.